docs/next.config.ts

```diff
@@ -127,6 +127,7 @@ export default withDocsInfra({
                   {
                     loader: require.resolve('@mui/internal-markdown/loader'),
                     options: {
+                      enableOpenInNewTab: true,
```

Comment:
I made it opt-in; the "Open in new tab" button in the demo toolbar is not shown by default.
But the `?scopedDemo` URL is supported regardless of this setting.

I don't know if the Core wants to have this enabled for the whole repo.
You can disable `enableOpenInNewTab` and enable it for specific pages like this:

```js
import * as pageProps from 'docs/data/material/components/drawers/drawers.md?muiMarkdown';

export default function Page() {
  return <MarkdownDocs {...pageProps} enableOpenInNewTab />;
}
```

This could be particularly useful for pages like https://mui.com/material-ui/react-drawer/

Let me know if you have a preference for this setting, and I'll update the PR accordingly.